sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/SqliteCompiler.kt

```diff
@@ -46,6 +47,7 @@ class SqliteCompiler {
       projectPath: String
   ): Status {
     try {
+      nameAllocators.put(parseContext.start.inputStream, NameAllocator())
```

Comment:
how does it leak? I clean up in `finally`

I want to avoid passing around a name allocator to any invocation on the column definition just because it seems like sloppy code and this seemed like the only way not to do that.